python/ray/data/_internal/planner/plan_udf_map_op.py

```diff
@@ -376,9 +412,13 @@ def _validate_batch_output(batch: Block) -> None:
 def _generate_transform_fn_for_map_batches(
     fn: UserDefinedFunction,
 ) -> MapTransformCallable[DataBatch, DataBatch]:
-    if inspect.iscoroutinefunction(fn):
-        # UDF is a callable class with async generator `__call__` method.
-        transform_fn = _generate_transform_fn_for_async_map(fn, _validate_batch_output)
+
+    if _is_async_udf(fn):
+        transform_fn = _generate_transform_fn_for_async_map(
+            fn,
+            _validate_batch_output,
+            max_concurrency=DEFAULT_ASYNC_BATCH_UDF_MAX_CONCURRENCY,
```

Comment:
consider renaming it to `asyncio_concurrency` to distinguish it from Core's ma_concurrency